### PR TITLE
[elm][haskell-servant] Fix issue 3256

### LINKED ISSUE #3256: [BUG][Elm] Root operation causes StringIndexOutOfBoundsException
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Generating client code for Elm (0.19) from an openapi.yaml specification with a root operation `/` causes a StringIndexOutOfBoundsException:

```
Exception in thread "main" java.lang.RuntimeException: Could not generate api file for 'Default'
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:666)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:922)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:396)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:60)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.base/java.lang.StringLatin1.charAt(StringLatin1.java:47)
	at java.base/java.lang.String.charAt(String.java:693)
	at org.openapitools.codegen.languages.ElmClientCodegen.lambda$postProcessOperationsWithModels$3(ElmClientCodegen.java:449)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.openapitools.codegen.languages.ElmClientCodegen.postProcessOperationsWithModels(ElmClientCodegen.java:449)
	at org.openapitools.codegen.DefaultGenerator.processOperations(DefaultGenerator.java:1175)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:552)
	... 3 more
```

##### openapi-generator version

@openapitools/openapi-generator-cli@0.0.14-4.0.2 (installed via NPM)

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0

info:
  title: Elm Root Bug
  version: 1.0.0

paths:
  /:
    get:
      summary: Root endpoint that causes generator bug
      responses:
        204:
          description: No response body
```

##### Command line used for generation

```shell
npx openapi-generator generate -i bug.yaml -g elm -o output
```

##### Steps to reproduce

1. Save sample spec as `bug.yaml`
2. `npm install -g openapi-generator-cli`
3. `npx openapi-generator generate -i bug.yaml -g elm -o output`

##### Related issues/PRs

I could not find a related issue, as this seems to be a bug specific to the Elm generator implementation.

##### Suggest a fix

[ElmClientCodegen.java:449](https://github.com/OpenAPITools/openapi-generator/blob/v4.0.2/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java#L449) contains a lambda with `str.charAt(0)` which fails on operations with a path of `/`. This should probably have a fallback on something like `"root"` or even an empty string. The Haskell generator produces an empty string and an operation named `"rootGet"`.

### PR DESCRIPTION
**Pull request by Elewyth**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language. @eriktim (Elm)

### Description of the PR
Fix #3256. Both the elm and haskell-servant generator had an issue handling root-operations (i.e. `/`). This PR should introduce as little change as possible only handling this specific case, without affecting any other logic.

### CODE REVIEW

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/elm/ElmClientCodegenTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2019 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/elm/ElmClientCodegenTest.java]: @Elewyth thanks for the PR. For new files, please remove line 3: https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/haskellservant/HaskellServantCodegenTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2019 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/haskellservant/HaskellServantCodegenTest.java]: Same as the comment above.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/elm/ElmClientCodegenTest.java (hunk 1 above)
Comment (Elewyth) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/elm/ElmClientCodegenTest.java]: A pleasure :). removed the line.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/haskellservant/HaskellServantCodegenTest.java (hunk 2 above)
Comment (Elewyth) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/haskellservant/HaskellServantCodegenTest.java]: Removed the line.

**Pull request merged**